viewer/viewer.js

```diff
@@ -274,6 +275,27 @@ if (Config.get("passwordSecret")) {
     if (internals.userNameHeader !== undefined) {
       if (req.headers[internals.userNameHeader] !== undefined) {
         var userName = req.headers[internals.userNameHeader];
+        // Auto-creation of users from headers
+        if (internals.userAutoCreateTmpl !== undefined) {
+
+           // Check to see if we need to auto-create.
+           Db.getUserCache(userName, function(err, suser) {
+              if (err) {return res.send("ERROR - getUser - user: " + userName + " err:" + err);}
+              if (!suser || !suser.found) {
+                 var nuser = JSON.parse(new Function("return `" + internals.userAutoCreateTmpl + "`;").call(req.headers));
```

Comment:
Can you explain what this is doing?  Do you have a sample value you are trying to use so we can understand better?

Reply:
Oops, I see the example in the PR request.  let me think about it

Reply:
Do you want me to move the creation into the block with the original getUserCache?  I figured the penalty for calling getUserCache was minimal given that it is just a hashtable lookup when its a hit.